  @Test
  public void thrownException_threwUnexpected_expectFailure() {
    Class<? extends Throwable> expected = SummerException.class;
    Class<? extends Throwable> unexpected = UnsupportedOperationException.class;
    CheckedProvider<String> provider =
        CheckedProviders.throwing(StringCheckedProvider.class, unexpected);
    String message =
        String.format(
            "exception thrown by <%s>(: |\n)Not true that <%s> is an instance of <%s>. "
                + "It is an instance of <%s>",
            quote(getThrowingProviderName(UnsupportedOperationException.class.getName())),
            quote(UnsupportedOperationException.class.getName()),
            quote(SummerException.class.getName()),
            quote(UnsupportedOperationException.class.getName()));

    expectWhenTesting().that(provider).thrownException().isInstanceOf(expected);
    assertThat(expect.getFailure()).hasMessageThat().matches(message);
  }

